src/worker/checkForUpdateWorker.js

```diff
@@ -24,36 +38,71 @@ setInterval(() => {
         const latest = JSON.parse(data);
 
         fs.readFile(playerVersionFilePath, (err, playerVersionData) => {
-          if (err) {
-            console.error("Error reading local version file:", err);
-            return;
-          }
-
-          const local = JSON.parse(playerVersionData);
-
-          if (latest.version > local.version) {
-            for (const file of latest.files) {
-              if (file.os === os) {
-                process.send?.({
-                  type: "player update",
-                  path: `${baseUrl}/player/${file.path}`,
-                  size: file.size,
-                });
-              }
-            }
-          }
+          const local = readLocalPlayerVersionFile(err, playerVersionData);
```

Comment:
It also would be better if we can use [`util.promisify()`](https://nodejs.org/dist/latest-v8.x/docs/api/util.html#util_util_promisify_original) for `fs.readFile()`, to replace the callback chain to promise based control flow.

Reply:
Thank you. It may be less stable, but I missed it.
I have completed the task and would appreciate your review.

Reply:
I had received approval, so I proceeded to merge the PR. 👀